### PR TITLE
feature: add support for HikariCP

### PR DESCRIPTION
**Pull request by diguage**
HikariCP is a solid, high-performance, JDBC connection pool. 

Spring Boot uses HikariCP as the default DataSource from 2.X. 

* [brettwooldridge/HikariCP: 光 HikariCP・A solid, high-performance, JDBC connection pool at last.](https://github.com/brettwooldridge/HikariCP)
* [MySQL Configuration · brettwooldridge/HikariCP Wiki](https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration)

### CODE REVIEW

File: server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        HikariConfig config = new HikariConfig(properties);
+        config.setDriverClassName(getDriverClassName());
+        config.setJdbcUrl(getUrl());
+        config.setUsername(getUser());
+        config.setPassword(getPassword());
+        config.setMaximumPoolSize(getMaxConn());
+        config.setMinimumIdle(getMinConn());
+        config.setAutoCommit(true);
+        config.setConnectionTestQuery(getValidationQuery(getDBType()));
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java:55]: image

**Comment by zjinlei** [comment]
@diguage don't use rebase to merge commit log

**Comment by diguage** [comment]
> @diguage don't use rebase to merge commit log
👌

If I update the code for some code reviews, how to merge the commit log?

**Comment by zjinlei** [comment]
> > @diguage don't use rebase to merge commit log
> > 👌
> 
> If I update the code for some code reviews, how to merge the commit log?
> 
> ```shell
> git pull diguage <BranchName>
> // update the code
> git commit -am "Commit log"
> git push diguage <BranchName>
> ```
> 
> Is it OK?

ok.
Squash merge will be used during the merge, and the incremental commit logs need to be viewed during the review.

**Comment by diguage** [comment]
> Squash merge will be used during the merge, and the incremental commit logs need to be viewed during the review.

Thank you. I will follow the advice.

File: server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java (hunk 1 above)
Comment (slievrly) [reply server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java:55]: @diguage How was the problem solved?

Comment (diguage) [reply server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java:55]: @slievrly 
 Yes. I delete the statement.

Comment (diguage) [reply server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java:55]: @slievrly 
 Yes. I have deleted the statement.

Comment (slievrly) [reply server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java:55]: @diguage Is the code not committed?

Comment (diguage) [reply server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java]: @slievrly 
Sorry. I forgot to commit.

**Comment by diguage** [comment]
> Squash merge will be used during the merge, and the incremental commit logs need to be viewed during the review.

@zjinlei 

Just now, I updated the code. But there is one more the commit log.

image

I executed the following commands:

```bash
git commit -am "delete config for connectionTestQuery"                                                                                     

git pull diguage hikari-data-source

git push diguage hikari-data-source
```

**Comment by zjinlei** [comment]
> > Squash merge will be used during the merge, and the incremental commit logs need to be viewed during the review.
> 
> @zjinlei
> 
> Just now, I updated the code. But there is one more the commit log.
> 
> image
> 
> I executed the following commands:
> 
> ```shell
> git commit -am "delete config for connectionTestQuery"                                                                                     
> 
> git pull diguage hikari-data-source
> 
> git push diguage hikari-data-source
> ```

no problem

File: server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        HikariConfig config = new HikariConfig(properties);
+        config.setDriverClassName(getDriverClassName());
+        config.setJdbcUrl(getUrl());
+        config.setUsername(getUser());
+        config.setPassword(getPassword());
+        config.setMaximumPoolSize(getMaxConn());
+        config.setMinimumIdle(getMinConn());
+        config.setAutoCommit(true);
+
```
[the remark this answers is not in the document]
Comment (diguage) [reply server/src/main/java/io/seata/server/store/HikariDataSourceGenerator.java]: OK

**Comment by a364176773** [comment]
由于前期Seata社区治理规范问题部分代码作者未签署CLA，可能引发社区知识产权风险问题。请所有在Seata社区贡献过代码（包含：主项目、官网、samples和多语言项目等）的 contributor 帮忙在这个链接登录github账号签署相应的开发者CLA，https://cla-assistant.io/seata/seata  。2023.1.31 未签署CLA的代码将会被rewrite，拜托大家帮忙签一下。

**Comment by diguage** [comment]
signed.

**Pull request merged**